src/test/java/com/hazelcast/kubernetes/DnsEndpointResolverTest.java

```diff
@@ -72,7 +73,23 @@ public void testValidServiceDns() throws Exception {
         assertEquals("127.0.0.1", nodes.get(0).getPrivateAddress().getHost());
     }
 
+    @Test
+    public void testDnsFailFlow() throws Exception {
+        DnsEndpointResolver endpointResolver = PowerMockito.spy(new DnsEndpointResolver(LOGGER, "hazelcast.com", SERVICE_DNS_TIMEOUT));
+        PowerMockito.when(endpointResolver, MemberMatcher.method(DnsEndpointResolver.class, "buildLookup", null)).withNoArguments().thenReturn(lookup);
+
+        when(lookup.getResult()).thenReturn(Lookup.HOST_NOT_FOUND);
+        when(lookup.run()).thenReturn(getRecords());
+
+        List<DiscoveryNode> nodes = endpointResolver.resolve();
+        assertTrue(nodes.isEmpty());
+
+        verify(lookup).run();
+        verify(lookup).getResult();
+        verify(lookup).getErrorString();
+    }
```

Comment:
I am not sure the last 3 `verify` statements are needed because `Lookup` is not our implementation

Reply:
Since, we check flow and our `DnsEndpointResolver` instantiates and calls `Lookup` as a crucial part of discovery, this flow check should be needed when DNS lookup fails to give proper error message.